pkg/crypto/keys.go

```diff
@@ -58,11 +37,5 @@ func GetPublicKeysFromPrivateKey(privateKey *ecdsa.PrivateKey) ([]byte, string,
 // GetPublicKeyFromBase58 Get public key ECDSA from public key base58
 func GetPublicKeyFromBase58(publicBase58 string) (*ecdsa.PublicKey, error) {
 	publicKeyBytes := base58.DecodeAlphabet(publicBase58, base58.BTCAlphabet)
-
-	curve := secp256k1.SECP256K1()
-	x, y := elliptic.Unmarshal(curve, publicKeyBytes)
-	if x == nil {
-		return nil, fmt.Errorf("invalid secp256k1 public key")
-	}
-	return &ecdsa.PublicKey{Curve: curve, X: x, Y: y}, nil
+	return ethcrypto.UnmarshalPubkey(publicKeyBytes)
```

Comment:
[this](https://github.com/ethereum/go-ethereum/blob/127ce93db4f5af4d7ada3066a15942a0ab59a948/crypto/crypto.go#L168) is a utility method that's only wrapping someother code

Reply:
I wonder if there's scope to inline it (copy/paste)

Reply:
(As above - uses `S256` which is a `go-ethereum/crypto` definition.) The point is that the general golang `crypto` methods don't implement `secp256k1`)